clients/instance/ibm-pi-workspaces.go

```diff
@@ -82,6 +82,9 @@ func (f *IBMPIWorkspacesClient) Create(name, location, groupID, plan string) (*r
 		return nil, nil, fmt.Errorf("workspace creation error, incorrect plan value; either \"public\" or \"private\" is allowed")
 	}
 	params := resourceController.NewCreateResourceInstanceOptions(name, location, groupID, planID)
+	if parameters != nil && len(parameters) > 0 {
```

Comment:
You can directly use len() as it handles nil values. 
```suggestion
	if len(parameters) > 0 {
```

Reply:
updated